### PR TITLE
Add Watchman support #72

### PR DESCRIPTION
**Pull request by Vinnie1991**
Installs Watchman support automatic with creating a new Ember project.

As asked in issue "Install watchman #72"

### CODE REVIEW

File: Ember Hearth/Watchman.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        var term = Terminal()
+        return term.runTerminalCommandSync("watchman -v")
+    }
+
+    static let name = "Watchman"
+    let name: String = Watchman.name
+
+    func install(completion:(success:Bool) -> ()) -> NSTask? {
+        var term = Terminal()
+        return term.runTerminalCommandAsync("brew install watchman", completion: { (result) -> () in
```
Comment (thomassnielsen) [inline Ember Hearth/Watchman.swift:31]: We should use NPM to install dependencies, since we can't depend on brew being on the system.

**Comment by thomassnielsen** [comment]
Thanks! I added a comment about using NPM over brew. The brew module is mainly ment for future use, updating any pre-installed dependencies that happened to be installed by brew. If you could update that I'll pull down the branch, test it, and most likely merge it.

**Comment by thomassnielsen** [comment]
Also, watchman doesn't have a `-v` flag, so we can't check install that way. I'm thinking we could check install by checking the path via `which`. For version number we can't do much as the watchman binary doesn't have it. We could ask NPM for the version, but that won't work if the user has it from before via brew or ports. So either just return `"?"` from version, or check NPM first, then return ? if it's installed but not listed in NPM.

**Comment by Vinnie1991** [comment]
OK, I'll update it.

**Comment by Vinnie1991** [comment]
Problem is: Watchman installed through NPM doesn't work with Ember-CLI.
You'll still get "Could not find watchman, falling back to NodeWatcher for file system events.".

As the Ember.js documentation already said: Watchman must be installed through Homebrew.
http://www.ember-cli.com/#watchman

**Comment by Vinnie1991** [comment]
Maybe an idea:

Let Ember-Heart check and/or install Homebrew and then install Watchman through Brew?

**Comment by thomassnielsen** [comment]
Oh, that's an interesting issue. I'll have to think a bit about that one, and get some input from @kimroen as well. If we need to install homebrew for the user, we may want to use that for more stuff as well. 

We tried hard to avoid depending on homebrew in the first place because it's pretty invasive, but we might not have a choice at this point.

**Comment by thomassnielsen** [comment]
I had a IRL chat with @kimroen about this. This is our current plan (feel free to suggest alternatives):
1. We'll install Watchman automatically during dependency setup _if you already have `brew`_
2. We're planning an event log system. We'll show the same kind of warning as Ember-CLI does if you don't have Watchman, but with a popup-GUI that lets you one-click install brew and Watchman

So for this PR, we need to implement Watchman as optional based on the status of brew. Specifically, it shouldn't even be listed during installation if you don't have brew (since we don't have a way to install it for you yet). I'd love if you could add that check. If not, I might be able to fix it this weekend.

**Comment by thomassnielsen** [comment]
Awesome. This looks good to me now. @Vinnie1991 are you planning do do more on this PR, or should I merge? (Asking since you haven't commented after the latest commit yet). I tested locally with and without brew, and it worked as expected.

**Comment by Vinnie1991** [comment]
Oh sorry @thomassnielsen. 
Maybe a next issue when you've decided that Brew will be used more. But for now, this is done.
You can merge the PR.

**Comment by thomassnielsen** [comment]
Once again, thanks for the PR. This will make it into 0.4.0. With some luck we'll have it baked into the event system #87 as mentioned above as well. :+1:

**Comment by Vinnie1991** [comment]
You're welcome. When you've another issue I'll help again :)
(First time I'm contributing to OS)

**Pull request merged**